server/svix-server/src/lib.rs

```diff
@@ -26,49 +26,52 @@ pub mod core;
 pub mod db;
 pub mod error;
 pub mod queue;
+pub mod redis;
 pub mod v1;
 pub mod webhook;
 pub mod worker;
 
 pub async fn run(cfg: Configuration, listener: Option<TcpListener>) {
     let pool = init_db(&cfg).await;
-    let redis_pool = if let Some(redis_dsn) = &cfg.redis_dsn {
-        tracing::debug!("Redis: Initializing pool");
-        let manager =
-            RedisConnectionManager::new(redis_dsn.to_string()).expect("Error initializing Redis");
-        Some(
-            bb8::Pool::builder()
-                .build(manager)
-                .await
-                .expect("Error initializing Redis"),
-        )
-    } else {
-        None
-    };
 
-    tracing::debug!("Queue type: {:?}", cfg.queue_type);
-    let (queue_tx, queue_rx) = match cfg.queue_type {
-        QueueType::Memory => queue::memory::new_pair().await,
-        QueueType::Redis => {
-            queue::redis::new_pair(redis_pool.clone().unwrap_or_else(|| {
-                panic!("Choosing queue_type=Redis requires setting a redis DSN.")
-            }))
-            .await
-        }
+    let redis_dsn = || {
+        cfg.redis_dsn
+            .as_ref()
+            .expect("Redis DSN not found")
+            .as_str()
     };
```

Comment:
This is a bit weird, why have it as a function if it's just borrow in both places?

Reply:
Putting the `expect` in one place instead of in 4 different spots was the goal.

Reply:
Why not have:
```suggestion
    let redis_dsn =
        cfg.redis_dsn
            .as_ref()
            .expect("Redis DSN not found")
            .as_str()
    ;
```
This puts it just one in one place, no?

Reply:
Because this will panic if `redis_dsn` is not defined.